tensordict/_torch_func.py

```diff
@@ -675,20 +675,32 @@ def stack_fn(key, values, is_not_init, is_tensor):
                 key: stack_fn(key, values, is_not_init, is_tensor)
                 for key, (values, is_not_init, is_tensor) in out.items()
             }
+
+            if (names := list_of_tensordicts[0]._has_names()) is not None:
+                names = list_of_tensordicts[0].names
+                names.insert(dim, None)
+
             result = clz._new_unsafe(
                 out,
                 batch_size=LazyStackedTensorDict._compute_batch_size(
                     batch_size, dim, len(list_of_tensordicts)
                 ),
+                names=names,
                 device=device,
             )
             if is_tc and not is_tensorclass(result):
                 return clz._from_tensordict(result)
             return result
         else:
+
+            if (names := list_of_tensordicts[0]._has_names()) is not None:
+                names = list_of_tensordicts[0].names
+                names.insert(dim, None)
+
             out = LazyStackedTensorDict(
```

Comment:
I think here lazy stack will preserve the names, you can add one name for the new dim though